casper/src/main/scala/coop/rchain/casper/Validate.scala

```diff
@@ -30,6 +32,14 @@ object Validate {
       "ed25519" -> Ed25519.verify
     )
 
+  def childMapIterator(blockHashSet: Set[BlockHash], internalMap: Map[BlockHash, BlockMessage]) =
+    new Iterator[BlockMessage] {
```

Comment:
I don't understand the point of defining your own iterator instance. Is this not the same as `blockHashSet.iterator.map(internalMap)`?